### PR TITLE
Documentation on the CI build system.

### PR DESCRIPTION
**Pull request by markmandel**
From #149 - realised we needed at least a note explaining how the build system works.

### CODE REVIEW

**Comment by pondohva** [comment]
There is nice example we can adopt for helm chart https://github.com/kubernetes/charts/tree/master/test/circle

For helm we need to have a:

1. CI system where we can run lint and package commands
2. storage (I prefer google cloud storage for this, also we can use github pages or smth else)

Also, we can contribute to kubernetes/charts repo agones helm chart from #149. This repo is a common place to look for helm charts, they provide CI system and 2 repos: https://kubernetes-charts.storage.googleapis.com and https://kubernetes-charts-incubator.storage.googleapis.com

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -31,5 +31,16 @@ Participation in this project comes under the [Contributor Covenant Code of Cond
 - Once review has occurred, please rebase your PR down to a single commit. This will ensure a nice clean Git history.
 - Finally - *Thanks* for considering submitting code to Agones!
 
+## Continuous Integration
```
Comment (pondohva) [inline CONTRIBUTING.md]: nice!

**Comment by pondohva** [comment]
We need to add go lint/go vet and tests for golang code also.

**Comment by markmandel** [comment]
@pondohva we run the go tests on every build, see [here](https://github.com/GoogleCloudPlatform/agones/blob/master/cloudbuild.yaml#L30) in the cloudbuild.yaml.

While [gometalinter](https://github.com/alecthomas/gometalinter) is part of the build image (and therefore shell), I'm reticent to make it part of the test/build pipeline, just because of the propensity for it to raise false positives, and also because it doesn't ignore generated files - of which there are quite a few in this project between gRPC and generated CRD apis. (Also see [this issue](https://github.com/alecthomas/gometalinter/issues/270)).

That being said, if you feel that there are places in which we can do a more on the testing front - or feel you have a way to work around the above issues, definitely  submit a PR to the cloudbuild.yaml and supporting test suite.

**Comment by Kuqd** [comment]
I agree with @pondohva we should have a CI for the helm package when #149 is closed.

**Comment by markmandel** [comment]
Any reason I can't get an approval on this PR though, so I can get it merged? :smile:

**Comment by markmandel** [comment]
Thanks!

**Pull request merged**